From 7a9782e6f4b9c422212f8b33671a49a20901d5d0 Mon Sep 17 00:00:00 2001
From: Patric Stout <truebrain@openttd.org>
Date: Sun, 24 Mar 2019 16:47:18 +0100
Subject: [PATCH 0478/1053] Revert: "Fix #7151: Hang when concurrently starting
 AIs in multiplayer, or with shift pressed."

This reverts commit b1e40b6b569c206d672f677a8e474a73ecf2173b.

This is a fix for a patch that will be reverted in the next commit.
---
 src/command.cpp     | 6 +++---
 src/command_type.h  | 1 -
 src/company_cmd.cpp | 4 ----
 3 files changed, 3 insertions(+), 8 deletions(-)

diff --git a/src/command.cpp b/src/command.cpp
index 72b3f4302e..70258aaa1d 100644
--- a/src/command.cpp
+++ b/src/command.cpp
@@ -283,7 +283,7 @@ static const Command _command_proc_table[] = {
 
 	DEF_CMD(CmdTurnRoadVeh,                                    0, CMDT_VEHICLE_MANAGEMENT    ), // CMD_TURN_ROADVEH
 
-	DEF_CMD(CmdPause,                    CMD_SERVER | CMD_NO_EST, CMDT_SERVER_SETTING        ), // CMD_PAUSE
+	DEF_CMD(CmdPause,                                 CMD_SERVER, CMDT_SERVER_SETTING        ), // CMD_PAUSE
 
 	DEF_CMD(CmdBuyShareInCompany,                              0, CMDT_MONEY_MANAGEMENT      ), // CMD_BUY_SHARE_IN_COMPANY
 	DEF_CMD(CmdSellShareInCompany,                             0, CMDT_MONEY_MANAGEMENT      ), // CMD_SELL_SHARE_IN_COMPANY
@@ -307,7 +307,7 @@ static const Command _command_proc_table[] = {
 	DEF_CMD(CmdChangeBankBalance,                      CMD_DEITY, CMDT_MONEY_MANAGEMENT      ), // CMD_CHANGE_BANK_BALANCE
 	DEF_CMD(CmdBuildCanal,                              CMD_AUTO, CMDT_LANDSCAPE_CONSTRUCTION), // CMD_BUILD_CANAL
 	DEF_CMD(CmdCreateSubsidy,                          CMD_DEITY, CMDT_OTHER_MANAGEMENT      ), // CMD_CREATE_SUBSIDY
-	DEF_CMD(CmdCompanyCtrl, CMD_SPECTATOR | CMD_CLIENT_ID | CMD_NO_EST, CMDT_SERVER_SETTING  ), // CMD_COMPANY_CTRL
+	DEF_CMD(CmdCompanyCtrl,        CMD_SPECTATOR | CMD_CLIENT_ID, CMDT_SERVER_SETTING        ), // CMD_COMPANY_CTRL
 	DEF_CMD(CmdCustomNewsItem,          CMD_STR_CTRL | CMD_DEITY, CMDT_OTHER_MANAGEMENT      ), // CMD_CUSTOM_NEWS_ITEM
 	DEF_CMD(CmdCreateGoal,              CMD_STR_CTRL | CMD_DEITY, CMDT_OTHER_MANAGEMENT      ), // CMD_CREATE_GOAL
 	DEF_CMD(CmdRemoveGoal,                             CMD_DEITY, CMDT_OTHER_MANAGEMENT      ), // CMD_REMOVE_GOAL
@@ -558,7 +558,7 @@ bool DoCommandP(TileIndex tile, uint32 p1, uint32 p2, uint32 cmd, CommandCallbac
 	bool estimate_only = _shift_pressed && IsLocalCompany() &&
 			!_generating_world &&
 			!(cmd & CMD_NETWORK_COMMAND) &&
-			!(GetCommandFlags(cmd) & CMD_NO_EST);
+			(cmd & CMD_ID_MASK) != CMD_PAUSE;
 
 	/* We're only sending the command, so don't do
 	 * fancy things for 'success'. */
diff --git a/src/command_type.h b/src/command_type.h
index 650a8987a4..e7512f11d2 100644
--- a/src/command_type.h
+++ b/src/command_type.h
@@ -395,7 +395,6 @@ enum CommandFlags {
 	CMD_CLIENT_ID = 0x080, ///< set p2 with the ClientID of the sending client.
 	CMD_DEITY     = 0x100, ///< the command may be executed by COMPANY_DEITY
 	CMD_STR_CTRL  = 0x200, ///< the command's string may contain control strings
-	CMD_NO_EST    = 0x400, ///< the command is never estimated.
 };
 DECLARE_ENUM_AS_BIT_SET(CommandFlags)
 
diff --git a/src/company_cmd.cpp b/src/company_cmd.cpp
index 8cc876751c..2c5a7dc467 100644
--- a/src/company_cmd.cpp
+++ b/src/company_cmd.cpp
@@ -724,10 +724,6 @@ void OnTick_Companies()
 		/* Allow multiple AIs to possibly start in the same tick. */
 		do {
 			if (!MaybeStartNewCompany()) break;
-
-			/* In networking mode, we can only send a command to start but it
-			 * didn't execute yet, so we cannot loop. */
-			if (_networking) break;
 		} while (AI::GetStartNextTime() == 0);
 	}
 
-- 
2.55.0

